src/app/pages/profile/financial-information/financial-information.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import {
+  UserProfileUpdateErrorResponse,
+  UserProfileUpdatedResponse
+} from 'src/app/models/Profile';
+import { Subscription } from 'rxjs';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { ProfileService } from 'src/app/services/profile/profile.service';
+import { first } from 'rxjs/operators';
+import { removeSubscription } from 'src/app/helpers/unsubscribe';
+
+@Component({
+  selector: 'app-financial-information',
+  templateUrl: './financial-information.component.html',
+  styleUrls: ['financial-information.component.scss']
+})
+export class FinancialInformationComponent implements OnInit, OnDestroy {
+  deposits: any;
+  cardInfo: any;
+  constructor(
+    private profileService: ProfileService,
+    private spinner: NgxSpinnerService
+  ) {}
+  subscribe: Subscription[] = [];
+  ngOnInit() {
+    this.getDeposits();
+    this.getCardInfo();
+  }
+
+  getDeposits() {
+    this.profileService
+      .getDeposits()
+      .pipe(first())
+      .subscribe(
+        response => {
+          response.data.transactions.map(transaction => {
+            if ( transaction.percentage_completion !== '100' ) {
+              this.deposits = transaction;
+            }
+            return this.deposits;
+          });
+        },
+        error => {
+          this.deposits = {error: 'deposits couldn\'t be found'};
+        }
+      );
+  }
+
+  getCardInfo() {
+    this.profileService.userProfile$.subscribe(response => {
+      const profile = response.data.profile;
+      if (Object.keys(profile.card_info.card_info).length === 0) {
+        this.cardInfo = null;
+      } else {
+        const cardNumberFull = [...profile.card_info.card_info.card_number];
+        this.cardInfo = {
+          cardOwner : `${profile.user.first_name} ${profile.user.last_name}`,
+          cardNumber: `${cardNumberFull[9]} ${cardNumberFull[10]} ${cardNumberFull[11]} ${cardNumberFull[12]}`,
+          cardExpiry: profile.card_info.card_info.card_expiry,
+        };
+        this.getCardBrand(profile.card_info.card_info.card_brand);
+      }
+    });
+  }
+
+  getCardBrand(brand) {
```

Comment:
What do you think of rewriting this function to 

```
  getCardBrand(brand) {
      const cardUrls = {
        VISA: 'https://upload.wikimedia.org/wikipedia/commons/thumb/0/04/Visa.svg/1200px-Visa.svg.png',
        'MASTER CARD': 'https://www.trzcacak.rs/myfile/full/213-2132174_mastercard-logo-png-circle.png',
        'AMERICAN EXPRESS': 'https://blockchainstocks.com/wp-content/uploads/2018/08/American-Express_Pentagram_Boteco-Design_04.jpg',
        'PAY PAL': 'https://www.imore.com/sites/imore.com/files/topic_images/2016/paypal-topic.png'
      };
      this.cardInfo.cardBrand = cardUrls[brand];
      return;
  }
```

It's much easier to read and also the `ts-lint` error of redundant local variable gets eradicated

Reply:
Thank you @Tevinthuku This is also a good approach. I have worked on it.